grails-app/domain/org/pih/warehouse/product/Product.groovy

```diff
@@ -457,12 +458,23 @@ class Product implements Comparable, Serializable {
     }
 
     /**
-     * Currently not implement since it would require coupling InventoryService to Product.
+     * Get quantity available to promise for this product given the location
      *
      * @param locationId
      */
-    def getQuantityAvailableToPromise(Integer locationId) {
-        throw new NotImplementedException()
+    def getQuantityAvailableToPromise(String locationId) {
```

Comment:
I think this should be in the ProductAvailabilityService with params `Location location` and `Product product`